src/wallet/message_store.cpp

```diff
@@ -0,0 +1,1389 @@
+    }
+  }
+  return true;
+}
+
+// Check if all members have a label set (as it's a requirement for starting auto-config
+// by the "manager")
+bool message_store::member_labels_complete() const
+{
+  for (size_t i = 0; i < m_members.size(); ++i) {
+    const coalition_member &m = m_members[i];
+    if (m.label.empty()) {
+      return false;
+    }
+  }
+  return true;
+}
+
+void message_store::get_member_config(std::string &member_config)
+{
+  std::stringstream oss;
+  boost::archive::portable_binary_oarchive ar(oss);
+  ar << m_members;
+  member_config = oss.str();
+}
+
+void message_store::unpack_member_config(const multisig_wallet_state &state, const std::string &member_config,
+                                         std::vector<coalition_member> &members)
+{
+  std::stringstream iss;
+  iss << member_config;
+  boost::archive::portable_binary_iarchive ar(iss);
+  ar >> members;
+}
+
+void message_store::process_member_config(const multisig_wallet_state &state, const std::string &member_config)
+{
+  // The members in "member_config" and the resident wallet members are matched not by label, but
+  // by Monero address, and ALL labels will be set from "member_config", even the "me" label. 
+  // In the auto-config process as implemented now the auto-config manager is responsible for defining
+  // the labels, and right at the end of the process ALL wallets use the SAME labels. The idea behind this
+  // is preventing problems like duplicate labels and confusion (Bob choosing a label "IamAliceHonest").
+  // (Of course members are free to re-define any labels they don't like AFTER auto-config.)
+  //
+  // Usually this method will be called with only the "me" member defined in the wallet, and may
+  // produce errors if that wallet contains additional members that have nothing to do with those arriving
+  // in "member_config".
+  std::vector<coalition_member> members;
+  unpack_member_config(state, member_config, members);
+  
+  uint32_t new_index = 1;
+  for (size_t i = 0; i < members.size(); ++i)
+  {
+    const coalition_member &m = members[i];
+    uint32_t index;
+    uint32_t take_index;
+    bool take = true;
+    bool found = get_member_index_by_monero_address(m.monero_address, index);
+    if (found)
+    {
+      // Redefine existing (probably "me", under usual circumstances)
+      take_index = index;
+    }
+    else
+    {
+      // Add new; neglect that we may erroneously overwrite already defined members
+      take_index = new_index;
+      new_index++;
+    }
+    if (take)
+    {
+      coalition_member &modify = m_members[take_index];
+      modify.label = m.label;  // ALWAYS set label, see comments above
+      if (!modify.me)
+      {
+        modify.transport_address = m.transport_address;
+        modify.monero_address_known = m.monero_address_known;
+        if (m.monero_address_known)
+        {
+          modify.monero_address = m.monero_address;
+        }
+      }
+    }
+  }
+  save(state);
+}
```

Comment:
Yes, it could, but only if the people don't use the underlying auto-config mechanism as intended and as it makes sense, or if they configure the same Monero address for more than one member.

Here and in similar places I currently go by the following: "So many things could go wrong in theory, to test them all would mean much work to program and large code. Let's see what will go wrong in practice and check *that* when improving the MMS."

I agree that the border between this and "I was just lazy" is quite thin :)